pkg/controller/migplan/validation.go

```diff
@@ -737,3 +748,63 @@ func containsAccessMode(modeList []kapi.PersistentVolumeAccessMode, accessMode k
 	}
 	return false
 }
+
+// Validate the pods, all should be healthy befor migration
+func (r ReconcileMigPlan) validatePods(plan *migapi.MigPlan) error {
+	if plan.Status.HasAnyCondition(Suspended) {
+		plan.Status.StageCondition(SourcePodsNotHealthy)
+		return nil
+	}
+
+	cluster, err := plan.GetSourceCluster(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	if cluster == nil || !cluster.Status.IsReady() {
+		return nil
+	}
+
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	unhealthyResources := migapi.UnhealthyResources{}
+	for _, ns := range plan.Spec.Namespaces {
+		unhealthyPods, err := health.PodsUnhealthy(client, &k8sclient.ListOptions{
+			Namespace: ns,
+		})
+		if err != nil {
+			log.Trace(err)
+			return err
+		}
+
+		workload := migapi.Workload{
+			Name: "Pods",
+		}
+		for _, unstrucredPod := range *unhealthyPods {
+			pod := &kapi.Pod{}
+			err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstrucredPod.UnstructuredContent(), pod)
+			if err != nil {
+				log.Trace(err)
+				return err
+			}
+
+			workload.Resources = append(workload.Resources, pod.Name)
+		}
+
+		if len(workload.Resources) != 0 {
+			unhealthyNamespace := migapi.UnhealthyNamespace{
+				Name:      ns,
+				Workloads: []migapi.Workload{workload},
+			}
+			unhealthyResources.Namespaces = append(unhealthyResources.Namespaces, unhealthyNamespace)
+		}
+	}
```

Comment:
Please set the `SourcePodsNotHealthy` condition when UnhealthyNamespaces is not empty.

Reply:
done